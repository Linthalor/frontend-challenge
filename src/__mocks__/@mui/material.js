const genMockComponent = (name) => (props) => {
  const MockName = `mocked-mui-${name}`;
  const {sx, ...remainingProps} = props;
  return <MockName {...remainingProps} sx={sx ? JSON.stringify(sx) : undefined} />;
}

module.exports = {
  ...jest.requireActual('@mui/material'),
  Box: genMockComponent('box'),
  Button: genMockComponent('button'),
  IconButton: genMockComponent('icon-button'),
  InputAdornment: genMockComponent('input-adornment'),
  TextField: genMockComponent('text-field'),
  Typography: genMockComponent('typography'),
  Card: genMockComponent('card'),
  CardContent: genMockComponent('card-content'),
  CardActions: genMockComponent('card-actions'),
};