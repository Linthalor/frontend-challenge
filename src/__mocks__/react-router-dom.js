module.exports = {
  ...jest.requireActual('react-router-dom'),
  Navigate: (props) => {
    const MockName = `mocked-react-router-dom-navigate`;
    return <MockName {...props} />;
  }
};