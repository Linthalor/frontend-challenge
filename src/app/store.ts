import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { signUpReducer } from '../features/sign-up/sign-up.slice';
import { colorReducer } from '../features/color/color.slice';

export const store = configureStore({
  reducer: {
    signUp: signUpReducer,
    color: colorReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
