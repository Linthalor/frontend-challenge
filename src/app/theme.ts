import { createTheme } from '@mui/material';

// Themed using Upgrade's palette
export const theme = createTheme({
  palette: {
    primary: {
      light: '#60c7a1',
      main: '#00a669',
      dark: '#008051',
    },
    secondary: {
      main: '#3f3d56',
      light: '#6bc1fe',
    },
    success: {
      main: '#03b57a',
      contrastText: '#ffffff',
    },
    error: {
      main: '#c83c4c',
      contrastText: '#ffffff',
    },
    text: {
      primary: '#3f465e',
      secondary: '#008051'
    }
  },
});