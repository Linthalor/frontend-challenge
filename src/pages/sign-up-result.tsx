import { Check, Error } from '@mui/icons-material';
import { Button, Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';
import { useAppDispatch } from '../app/hooks';
import { SignUpCard } from '../components/signup-card';
import { resetSignUp } from '../features/sign-up/sign-up.slice';

export const SignupResult = ({ success }: { success?: true }) => {
  const dispatch = useAppDispatch();
  const { t } = useTranslation();

  const restart = () => {
    dispatch(resetSignUp());
  };

  return <>
    <SignUpCard
      content={<>
        <Typography
          sx={{
            textAlign: 'center',
            fontWeight: 'bold'
          }}
          variant="h5"
          color="text.primary"
          gutterBottom
        >
          {t(`page:signUpResult:${success ? 'success' : 'error'}:label`)}
        </Typography>
        <Typography sx={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
          {success 
            ? <Check fontSize="large" color="primary" sx={{ mr: 1 }} />
            : <Error fontSize="large" color="error" sx={{ mr: 1 }} />
          }
          {t(`page:signUpResult:${success ? 'success' : 'error'}:message`)}
        </Typography>
      </>}
      actions={<>
        <Button variant="contained" type="submit" onClick={restart}>
          {t('common:restart')}
        </Button>
      </>}
    />
  </>;
};