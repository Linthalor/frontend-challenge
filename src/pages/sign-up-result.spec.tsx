import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createTestStore, store as realStore } from '../app/store';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from '@mui/material';
import { theme } from '../app/theme';
import { SignupResult } from './sign-up-result';

jest.mock('@mui/material');

describe('SignupResult', () => {
  let store: typeof realStore;
  beforeEach(() => {
    store = createTestStore();
  });
  afterEach(() => {
    cleanup();
  });
  
  it('renders as expected on success', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <BrowserRouter>
            <SignupResult success />
          </BrowserRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders as expected on failure', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <BrowserRouter>
            <SignupResult />
          </BrowserRouter>
        </Provider>
      </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});