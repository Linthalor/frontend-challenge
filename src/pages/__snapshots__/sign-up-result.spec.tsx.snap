// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupResult renders as expected on failure 1`] = `
<mocked-mui-box
  sx="{\\"width\\":\\"100%\\",\\"height\\":\\"100%\\",\\"display\\":\\"flex\\",\\"justifyContent\\":\\"center\\",\\"alignItems\\":\\"center\\"}"
>
  <mocked-mui-card
    sx="{\\"width\\":400,\\"height\\":\\"unset\\",\\"display\\":\\"flex\\",\\"flexDirection\\":\\"column\\",\\"justifyContent\\":\\"center\\"}"
  >
    <mocked-mui-card-content
      sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"column\\",\\"alignItems\\":\\"stretch\\",\\"justifyContent\\":\\"start\\",\\"gap\\":1,\\"flex\\":1}"
    >
      <mocked-mui-typography
        color="text.primary"
        gutterbottom="true"
        sx="{\\"textAlign\\":\\"center\\",\\"fontWeight\\":\\"bold\\"}"
        variant="h5"
      >
        Error!
      </mocked-mui-typography>
      <mocked-mui-typography
        sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"row\\",\\"alignItems\\":\\"center\\"}"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeLarge css-104vmwh-MuiSvgIcon-root"
          data-testid="ErrorIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
          />
        </svg>
        Uh oh, something went wrong. Please try again later.
      </mocked-mui-typography>
    </mocked-mui-card-content>
    <mocked-mui-card-actions
      sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"row\\",\\"justifyContent\\":\\"end\\",\\"gap\\":1}"
    >
      <mocked-mui-button
        type="submit"
        variant="contained"
      >
        Restart
      </mocked-mui-button>
    </mocked-mui-card-actions>
  </mocked-mui-card>
</mocked-mui-box>
`;

exports[`SignupResult renders as expected on success 1`] = `
<mocked-mui-box
  sx="{\\"width\\":\\"100%\\",\\"height\\":\\"100%\\",\\"display\\":\\"flex\\",\\"justifyContent\\":\\"center\\",\\"alignItems\\":\\"center\\"}"
>
  <mocked-mui-card
    sx="{\\"width\\":400,\\"height\\":\\"unset\\",\\"display\\":\\"flex\\",\\"flexDirection\\":\\"column\\",\\"justifyContent\\":\\"center\\"}"
  >
    <mocked-mui-card-content
      sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"column\\",\\"alignItems\\":\\"stretch\\",\\"justifyContent\\":\\"start\\",\\"gap\\":1,\\"flex\\":1}"
    >
      <mocked-mui-typography
        color="text.primary"
        gutterbottom="true"
        sx="{\\"textAlign\\":\\"center\\",\\"fontWeight\\":\\"bold\\"}"
        variant="h5"
      >
        Success!
      </mocked-mui-typography>
      <mocked-mui-typography
        sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"row\\",\\"alignItems\\":\\"center\\"}"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-2s4ihf-MuiSvgIcon-root"
          data-testid="CheckIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
        You should receive a confirmation email soon.
      </mocked-mui-typography>
    </mocked-mui-card-content>
    <mocked-mui-card-actions
      sx="{\\"display\\":\\"flex\\",\\"flexDirection\\":\\"row\\",\\"justifyContent\\":\\"end\\",\\"gap\\":1}"
    >
      <mocked-mui-button
        type="submit"
        variant="contained"
      >
        Restart
      </mocked-mui-button>
    </mocked-mui-card-actions>
  </mocked-mui-card>
</mocked-mui-box>
`;
